### DISCUSSION TITLE
Experimental use of LLMs for completing issues

### DISCUSSION THREAD

**Discussion by adamamer20** [projectmesa/mesa-frames#121] [Ideas]
I believe with a good CI suite in place and a good template for issues (eg. current behavior, expected behavior...) we could use a local LLM like DeepSeek-Coder to solve the easiest issues and simply review the changes. This would aid development greatly because it would allow to focus only on the most difficult issues and less on actual mantainance. 
Look at this repo: https://github.com/princeton-nlp/SWE-agent?tab=readme-ov-file

**Comment by rht** [comment]
According to OpenAI's [modification of SWE-bench](https://openai.com/index/introducing-swe-bench-verified/), Aider performs better than SWE-Agent, while [Agentless](https://github.com/OpenAutoCoder/Agentless) is the best performing one.

**Comment by adamamer20** [comment]
> According to OpenAI's [modification of SWE-bench](https://openai.com/index/introducing-swe-bench-verified/), Aider performs better than SWE-Agent, while [Agentless](https://github.com/OpenAutoCoder/Agentless) is the best performing one.

Thanks @rht! Always on top :). I took a look at Agentless and right now the documentation is fairly lacking (it shows how to reproduce the SWE-Bench result but that's it). We can try using Aider, it seems well documented and easy to setup!
